tests/CacheTest.php

```diff
@@ -20,92 +22,131 @@ public function setUp()
         $this->registrar->forgetCachedPermissions();
 
         DB::connection()->enableQueryLog();
+    }
 
-        $this->assertCount(0, DB::getQueryLog());
+    /** @test */
+    public function it_can_cache_the_permissions()
+    {
+        $this->registrar->getPermissions();
 
-        $this->registrar->registerPermissions();
+        $this->assertQueryCount(self::QUERIES_PER_CACHE_PROVISION);
 
-        $this->assertCount(2, DB::getQueryLog());
+        $this->registrar->getPermissions();
 
-        DB::flushQueryLog();
+        $this->assertQueryCount(self::QUERIES_PER_CACHE_PROVISION);
     }
 
     /** @test */
-    public function it_can_cache_the_permissions()
+    public function it_flushes_the_cache_when_creating_a_permission()
     {
-        $this->registrar->registerPermissions();
+        app(Permission::class)->create(['name' => 'new']);
+
+        $this->resetQueryCount();
 
-        $this->assertCount(0, DB::getQueryLog());
+        $this->registrar->getPermissions();
+
+        $this->assertQueryCount(self::QUERIES_PER_CACHE_PROVISION);
     }
 
     /** @test */
-    public function permission_creation_and_updating_should_flush_the_cache()
+    public function it_flushes_the_cache_when_updating_a_permission()
     {
         $permission = app(Permission::class)->create(['name' => 'new']);
-        $this->assertCount(1, DB::getQueryLog());
 
-        $this->registrar->registerPermissions();
-        $this->assertCount(3, DB::getQueryLog());
+        $this->resetQueryCount();
```

Comment:
Isn't this `reset` a duplicate of line 61 below?

Reply:
Yep, you're right!